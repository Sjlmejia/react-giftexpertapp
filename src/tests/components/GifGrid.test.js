import React from 'react';
import {shallow} from 'enzyme'
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');
describe('test para GifGrid component', () => {
  const category = 'Iron Man';

  test('debe crearse gifgrid', () => {
    useFetchGifs.mockReturnValue({
      data:[],
      loading: true
    });
    const wrapp = shallow(<GifGrid category={category}/>);
    expect(wrapp).toMatchSnapshot();
  });

  test('debe de mostrar items cuando se carga imagenes UseFetchGifs', () => {
    useFetchGifs.mockReturnValue({
      data:[{
        id:1,
        url: 'https://local.com',
        title: 'cualquier cosa'
      }],
      loading: false
    });
    const wrapp = shallow(<GifGrid category={category}/>);
    expect(wrapp.find('p').exists()).toBe(false);
    expect(wrapp.find('GifGridItem').length).toBe(1);
  });
});