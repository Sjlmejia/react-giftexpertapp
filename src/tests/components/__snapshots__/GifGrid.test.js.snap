// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test para GifGrid component debe crearse gifgrid 1`] = `
<Fragment>
  <h3>
    Iron Man
  </h3>
  <p
    className="card"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
